code/lib/router/src/router.tsx

```diff
@@ -86,30 +105,55 @@ Location.displayName = 'QueryLocation';
 // A render-prop component for rendering when a certain path is hit.
 // It's immensely similar to `Location` but it receives an addition data property: `match`.
 // match has a truthy value when the path is hit.
-export const Match = ({ children, path: targetPath, startsWith = false }: MatchProps) => (
-  <Location>
-    {({ path: urlPath, ...rest }) =>
-      children({
-        match: getMatch(urlPath, targetPath, startsWith),
-        ...rest,
-      })
-    }
-  </Location>
-);
+function Match(props: MatchPropsStartsWith): ReactElement;
+function Match(props: MatchPropsDefault): ReactElement;
+function Match({
+  children,
+  path: targetPath,
+  startsWith = false,
+}: MatchPropsStartsWith | MatchPropsDefault) {
+  return (
+    <Location>
+      {({ path: urlPath, ...rest }) =>
+        children({
+          match: getMatch(urlPath, targetPath, startsWith),
+          ...rest,
+        })
+      }
+    </Location>
+  );
+}
 Match.displayName = 'QueryMatch';
 
 // A component to conditionally render children based on matching a target path
-export const Route = ({ path, children, startsWith = false, hideOnly = false }: RouteProps) => (
-  <Match path={path} startsWith={startsWith}>
-    {({ match }) => {
-      if (hideOnly) {
-        return <ToggleVisibility hidden={!match}>{children}</ToggleVisibility>;
-      }
-      return match ? children : null;
-    }}
-  </Match>
-);
+function Route(props: RoutePropsDefault): ReactElement;
+function Route(props: RoutePropsStartsWith): ReactElement;
+function Route(input: RoutePropsDefault | RoutePropsStartsWith) {
+  let propsA: RoutePropsStartsWith = input as RoutePropsStartsWith;
+  let propsB: RoutePropsDefault = input as RoutePropsDefault;
+
+  if (isRoutePropsDefault(input)) {
+    propsB = { ...input };
+    propsB.startsWith = false;
+  } else {
+    propsA = input;
+  }
+
+  const { children, hideOnly = false } = propsA || propsB;
+  return (
+    <Match {...(propsA || propsB)}>
```

Comment:
should `children and hideOnly` also be passed to `Match`?

Reply:
`children` **is** passed
`hideOnly` should not be passed, it's used to wrap children, hiding it, instead of removing it from DOM.

see:
https://github.com/storybookjs/storybook/blob/ad6103a3c054ed55ad986816406ac6cf582f0efd/code/lib/router/src/router.tsx#L144-L151

Reply:
So both properties shouldn't be passed to "Match" then?

```tsx
const { children, hideOnly = false, ...rest } = propsA || propsB;
  return (
    <Match {...rest}>...</Match>
```

Reply:
which is why they aren't...

I'm confused @valentinpalkovic what is it you'd like to see changed?